lib/ansible/modules/cloud/amazon/rds_instance.py

```diff
@@ -0,0 +1,1152 @@
+            a DB instance the RDS API defaults this to false if unspecified.
+        type: bool
+    db_cluster_identifier:
+        description:
+          - The DB cluster (lowercase) identifier to add the aurora DB instance to. The identifier must contain from 1 to
+            63 letters, numbers, or hyphens and the first character must be a letter and may not end in a hyphen or
+            contain consecutive hyphens.
+        aliases:
+          - cluster_id
+    db_instance_class:
+        description:
+          - The compute and memory capacity of the DB instance, for example db.t2.micro.
+        aliases:
+          - class
+          - instance_type
+    db_instance_identifier:
+        description:
+          - The DB instance (lowercase) identifier. The identifier must contain from 1 to 63 letters, numbers, or
+            hyphens and the first character must be a letter and may not end in a hyphen or contain consecutive hyphens.
+        aliases:
+          - instance_id
+          - id
+        required: True
+    db_name:
+        description:
+          - The name for your database. If a name is not provided Amazon RDS will not create a database.
+    db_parameter_group_name:
+        description:
+          - The name of the DB parameter group to associate with this DB instance. When creating the DB instance if this
+            argument is omitted the default DBParameterGroup for the specified engine is used.
+    db_security_groups:
+        description:
+          - (EC2-Classic platform) A list of DB security groups to associate with this DB instance.
+        type: list
+    db_snapshot_identifier:
+        description:
+          - The identifier for the DB snapshot to restore from if using I(creation_source=snapshot).
+    db_subnet_group_name:
+        description:
+          - The DB subnet group name to use for the DB instance.
+        aliases:
+          - subnet_group
+    domain:
+        description:
+          - The Active Directory Domain to restore the instance in.
+    domain_iam_role_name:
+        description:
+          - The name of the IAM role to be used when making API calls to the Directory Service.
+    enable_cloudwatch_logs_exports:
+        description:
+          - A list of log types that need to be enabled for exporting to CloudWatch Logs.
+        aliases:
+          - cloudwatch_log_exports
+        type: list
+    enable_iam_database_authentication:
+        description:
+          - Enable mapping of AWS Identity and Access Management (IAM) accounts to database accounts.
+            If this option is omitted when creating the cluster, Amazon RDS sets this to False.
+        type: bool
+    enable_performance_insights:
+        description:
+          - Whether to enable Performance Insights for the DB instance.
+        type: bool
+    engine:
+        description:
+          - The name of the database engine to be used for this DB instance. This is required to create an instance.
```

Comment:
I could pull it from https://github.com/boto/botocore/blob/c632931a6cc5eab4113c976d430bcb9c059f829f/botocore/data/rds/2014-09-01/service-2.json#L3045? It doesn't look like boto3 or botocore ever validate or do anything to the engine other than specify it to be a string. They just pass along the value. We could do that too. But then the user won't know what the valid values are without looking it up. I would be okay with removing `choices` from this option in the argspec (I'd have to move the documented choices to the description) and add the list as accepted (last known and in need of occassional-updating) enginges to the error message in case it is a bad value. That wouldn't impede unattended growth. Sound okay?

Reply:
Yeah, maybe you could add to the documentation on the option the current engines, but remove the `choices` validation so that new engines will still work but may be undocumented for up to 1 release.

Reply:
Addressed, also added a helpful error for invalid engines